### PR TITLE
fix: 1:1 chat concerns

### PR DESCRIPTION
**Pull request by BalogunofAfrica**
Fixes 
https://github.com/status-im/status-mobile/issues/17539 
https://github.com/status-im/status-mobile/issues/17912
https://github.com/status-im/status-mobile/issues/17913

### Summary
Fixes the design concerns in https://github.com/status-im/status-mobile/issues/17539, and also implements the changes to the context drawer for 1:1 chat seen here: https://www.figma.com/file/7KIYbhoqNGAIFonE0w9TDz/Messages-for-Mobile?type=design&node-id=1447-349773&mode=design&t=faJfVMrRatdEVuv1-0.

#### Platforms

- Android
- iOS

##### Functional

- 1-1 chats

status: ready

### CODE REVIEW

**Comment by BalogunofAfrica** [comment]
The `remove-chat-test` is failing, currently looking into it

**Review by Francesca-G — APPROVED** [review]
Here's the design [review](https://www.figma.com/file/Tf5nfkYvpbnNCo4rKLK7lS/Feedback-for-Mobile?type=design&node-id=7426%3A45659&mode=design&t=j5uaUZkHBvqF251V-1) :)

**Comment by BalogunofAfrica** [comment]
Thanks for your review, @Francesca-G . I'd create a follow up issue for the sticky user details

File: src/status_im/integration_test.cljs
Comment (flexsurfer) [inline src/status_im/integration_test.cljs]: could you pls elaborate on why we rename delete/remove to close

File: src/status_im/ui/screens/chat/sheets.cljs
Comment (flexsurfer) [inline src/status_im/ui/screens/chat/sheets.cljs]: feel free to remove anything which is not used anymore

File: src/status_im2/contexts/chat/events.cljs
Comment (flexsurfer) [inline src/status_im2/contexts/chat/events.cljs]: could you elaborate on this change, it looks pretty critical

**Review by flexsurfer — CHANGES_REQUESTED** [review]

File: src/status_im/integration_test.cljs
Comment (BalogunofAfrica) [reply src/status_im/integration_test.cljs]: The rename is due to the copy update from the designs: https://www.figma.com/file/7KIYbhoqNGAIFonE0w9TDz/Messages-for-Mobile?type=design&node-id=1447-349773&mode=design&t=OtI9WtlvD7M56ztS-0

File: src/status_im2/contexts/chat/events.cljs
Comment (BalogunofAfrica) [reply src/status_im2/contexts/chat/events.cljs]: The requirement for the destructive actions in the context drawer for 1:1 chat was changed. One of the actions was a delete message, but it was changed to close message where the message is just removed from the chat list while still retaining history.

This was communicated here: https://discord.com/channels/624634427930312714/928625369542713396/1167063940816179271

File: src/status_im/integration_test.cljs
Comment (flexsurfer) [reply src/status_im/integration_test.cljs]: yeah, but it doesn't mean we should rename it in the code base, copy is enough, its still should have same functionality, or?

File: src/status_im2/contexts/chat/events.cljs
Comment (flexsurfer) [reply src/status_im2/contexts/chat/events.cljs]: mhm i think we already had something similar cc @churik

File: src/status_im/integration_test.cljs
Comment (BalogunofAfrica) [reply src/status_im/integration_test.cljs]: The functionality changed, here's a link to the discuss:

https://discord.com/channels/624634427930312714/928625369542713396/1167063940816179271

File: src/status_im2/contexts/chat/events.cljs
Comment (flexsurfer) [reply src/status_im2/contexts/chat/events.cljs]: and we have like `active?` flag or something

Comment (flexsurfer) [reply src/status_im2/contexts/chat/events.cljs]: actually i looked at the code, and it seems like there should be no changes only copy should be changed, because it already works as expected

Comment (BalogunofAfrica) [reply src/status_im2/contexts/chat/events.cljs]: 🤔 The current implementation deletes the history alongside removing it from the home chat

File: translations/en.json
Comment (flexsurfer) [inline translations/en.json]: we already have `"confirm": "Confirm",`

File: src/status_im2/contexts/chat/events.cljs
Comment (flexsurfer) [reply src/status_im2/contexts/chat/events.cljs]: i don't see it removes history, it just clear it in react, we still have it in status-go

Comment (BalogunofAfrica) [reply src/status_im2/contexts/chat/events.cljs]: I don't see the history after I delete it and try chatting with the other device again, even after closing the app and re-opening 🤔

Comment (flexsurfer) [reply src/status_im2/contexts/chat/events.cljs]: so there are two parts where we have data, react and status-go, in your PR as soon as you relogin chat will be back, im not sure that is expected, so i would say current version in develop works as expected

File: translations/fa.json
Comment (flexsurfer) [inline translations/fa.json:426]: @jinhojang6 what do you think, is it better to keep key or its better to rename key as well in such cases?

File: src/status_im2/contexts/chat/events.cljs
Comment (BalogunofAfrica) [reply src/status_im2/contexts/chat/events.cljs]: I am not sure I understand 🤔, here's my experience with `develop`:

https://github.com/status-im/status-mobile/assets/45393944/c560e056-f706-4b39-b0eb-d51846fcf900

I don't get to see the messages when I go back to chat with the user who I closed(deleted) their chat, but the second point in the new requirement noted that we should be able to see their chat history, just that it would be removed from the list of chats

Comment (flexsurfer) [reply src/status_im2/contexts/chat/events.cljs]: oh right, probably we have something like deleted-at and we don't load messages, this should be changed then, it's more complicated changes needed, we need to see how deactivated method works in status-go

Comment (BalogunofAfrica) [reply src/status_im2/contexts/chat/events.cljs]: @flexsurfer Updated the `DeactivateChat` method from `status-go`.

Link to that PR here: https://github.com/status-im/status-go/pull/4324

**Review by flexsurfer — APPROVED** [review]

**Comment by BalogunofAfrica** [comment]
@Francesca-G Please help re-review :).

I have addressed the design concern here https://www.figma.com/file/Tf5nfkYvpbnNCo4rKLK7lS?node-id=7426:45659&mode=design#604878622

**Review by Francesca-G — COMMENTED** [review]
@BalogunofAfrica the message context drawer seems fixed ✨
The pressed state of the chat is closer to design but we're missing the rounded corners:

Screenshot 2023-11-22 alle 15 33 33


it shouldn't go all the way to the edge of the screen :)

**Comment by VolodLytvynenko** [comment]
Could you rebase please the go branch https://github.com/status-im/status-go/tree/fix/deactivate-chat  to the last develope ?

**Comment by BalogunofAfrica** [comment]
Hi @VolodLytvynenko I have updated the branch

**Comment by VolodLytvynenko** [comment]
hi @BalogunofAfrica thank you for PR. One issue is found:

The issue regarding the incorrect bell icon has been fixed for 1-1 chats. However, the wrong bell icon is still appearing both for channels and communities.

## ISSUE 1: Incorrect bell Icon displayed for communities and community channels
### Steps:
1. Create/join at least one community
2. Go to community list -> long tap the community
3. Open community -> Long tap the channel inside

### Actual result:
image

### Expected result:
The bell icon should display as crossed out while the community or channels are unmuted

**Comment by VolodLytvynenko** [comment]
@BalogunofAfrica Integration tests failed. Could you take a look at them please? https://ci.status.im/blue/organizations/jenkins/status-mobile%2Fprs%2Ftests/detail/PR-17795/18/pipeline/26

**Comment by BalogunofAfrica** [comment]
Hi @VolodLytvynenko I've addressed both issues

**Comment by churik** [comment]
@BalogunofAfrica please, do not forget to squash my commit too while merging, thanks!

**Comment by VolodLytvynenko** [comment]
@BalogunofAfrica Thank you for PR.  No issues from my side.

@Francesca-G could you take a look at this PR?

**Review by Francesca-G — APPROVED** [review]
Looks good ✨

Only issue to fix is the alignment between the username, contact icon and time:

Screenshot 2023-12-04 alle 09 28 28

as commented here: https://www.figma.com/file/Tf5nfkYvpbnNCo4rKLK7lS/Feedback-for-Mobile?type=design&node-id=7733%3A41550&mode=design&t=EXSvBBXLXok3UySO-1

**Comment by VolodLytvynenko** [comment]
Hey @BalogunofAfrica. Please don't merge the current PR. We've encountered an issue on the status go side (https://github.com/status-im/status-mobile/issues/18092) merging this PR affects our mobile develop

**Comment by VolodLytvynenko** [comment]
Hey @BalogunofAfrica  could you rebase status go https://github.com/status-im/status-go/tree/fix/deactivate-chat and current PR. I quickly rerun tests once more before merging this PR. Thanx!

**Comment by BalogunofAfrica** [comment]
Hi @VolodLytvynenko I have updated the PR

**Comment by VolodLytvynenko** [comment]
@BalogunofAfrica thank you for your work. PR is ready to be merged

**Pull request merged**
